src/couch/src/couch_ejson_size.erl

```diff
@@ -53,3 +62,130 @@ ejson_size(false) ->
 
 ejson_size(Atom) when is_atom(Atom) ->
     ejson_size(atom_to_binary(Atom, utf8)).
+
+
+utf8_string_size(Binary) ->
+    %% 2 is for open and closing "
+    utf8_string_size(Binary, 2).
+
+utf8_string_size(<<0:1, C:7/integer, Rest/binary>>, Acc) ->
```

Comment:
This parses over each character, I'd be worries about performance there. We'd have to measure though.